src/status_im/contexts/wallet/common/utils/networks.cljs

```diff
@@ -1,68 +1,20 @@
   (:require
     [clojure.string :as string]
-    [quo.foundations.resources :as resources]
-    [status-im.constants :as constants]
+    [status-im.contexts.wallet.networks.core :as networks]
     [utils.money :as money]
     [utils.number]))
 
-(def ^:private last-comma-followed-by-text-to-end-regex #",\s(?=[^,]+$)")
-(def ^:private max-network-prefixes 2)
-
-(def id->network
-  {constants/ethereum-mainnet-chain-id       constants/mainnet-network-name
-   constants/ethereum-sepolia-chain-id       constants/mainnet-network-name
-   constants/optimism-mainnet-chain-id       constants/optimism-network-name
-   constants/optimism-sepolia-chain-id       constants/optimism-network-name
-   constants/arbitrum-mainnet-chain-id       constants/arbitrum-network-name
-   constants/arbitrum-sepolia-chain-id       constants/arbitrum-network-name
-   constants/base-mainnet-chain-id           constants/base-network-name
-   constants/base-sepolia-chain-id           constants/base-network-name
-   constants/status-network-sepolia-chain-id constants/status-network-name})
-
-(defn- get-chain-id
-  [{:keys [mainnet-chain-id sepolia-chain-id testnet-enabled?]}]
-  (if testnet-enabled?
-    sepolia-chain-id
-    mainnet-chain-id))
-
 (defn network->chain-id
-  ([db network]
-   (let [{:keys [test-networks-enabled?]} (:profile/profile db)]
-     (network->chain-id {:network          network
-                         :testnet-enabled? test-networks-enabled?})))
-  ([{:keys [network testnet-enabled?]}]
-   (condp contains? (keyword network)
-     #{constants/mainnet-network-name (keyword constants/mainnet-short-name)}
-     (get-chain-id
-      {:mainnet-chain-id constants/ethereum-mainnet-chain-id
-       :sepolia-chain-id constants/ethereum-sepolia-chain-id
-       :testnet-enabled? testnet-enabled?})
-
-     #{constants/optimism-network-name (keyword constants/optimism-short-name)}
-     (get-chain-id
-      {:mainnet-chain-id constants/optimism-mainnet-chain-id
-       :sepolia-chain-id constants/optimism-sepolia-chain-id
-       :testnet-enabled? testnet-enabled?})
-
-     #{constants/arbitrum-network-name (keyword constants/arbitrum-short-name)}
-     (get-chain-id
-      {:mainnet-chain-id constants/arbitrum-mainnet-chain-id
-       :sepolia-chain-id constants/arbitrum-sepolia-chain-id
-       :testnet-enabled? testnet-enabled?})
-
-     #{constants/base-network-name (keyword constants/base-short-name)}
-     (get-chain-id
-      {:mainnet-chain-id constants/base-mainnet-chain-id
-       :sepolia-chain-id constants/base-sepolia-chain-id
-       :testnet-enabled? testnet-enabled?})
-
-     #{constants/status-network-name (keyword constants/status-network-short-name)}
-     (get-chain-id
-      ;; TODO: Update mainnet chain ID when Status Network Mainnet is launched
-      {:mainnet-chain-id 0
-       :sepolia-chain-id constants/status-network-sepolia-chain-id
-       :testnet-enabled? testnet-enabled?}))))
+  ([db network-name]
```

Comment:
I think this should be 2 separate functions because they are not really doing the same. The first one is accessing networks from database and the second one receives chain-id. First one works with db and second one works with network structure, so eventually they can be in different files.

Reply:
yeah good point, that function always looked weird :+1: